```{r}
library(tidyverse)
library(karyoploteR)
```

Height
```{r}
### joint
output_height = read.table("./saige/saige_output_height.txt", header = TRUE)
GR_input = output_height %>% mutate(chr = paste0("chr", CHR), start = POS, end = POS) %>% 
  select(chr, start, end, A1 = Allele1, A2 = Allele2, P = p.value)
saige_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)

### meta-analysis
df = read.table("/hpc/group/ochoalab/tt207/meta_analysis_aim/METAL/samafs/output_metal_sex_height1.txt", header = TRUE) 
df_clean = merge(output_height %>% select(CHR, POS, MarkerName = MarkerID), df, by = "MarkerName")
GR_input = df_clean %>% 
  mutate(chr = paste0("chr", CHR), start = as.numeric(POS), end = as.numeric(POS)) %>% 
  select(-POS) %>% select(chr, start, end, A1 = Allele1, A2 = Allele2, P = P.value)
sex_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)
```


```{r  fig.width=18, fig.height=7}

#png( '/hpc/group/ochoalab/tt207/meta_analysis_aim/samafs/plots/height_man.png', width=18.5, height=13, res=1000, units = 'in')
#par(mfrow=c(5,3), mar=c(2,2,2,2))
chr_plot = c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", 
             "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22")

pp <- getDefaultPlotParams(plot.type=4)
pp$ideogramlateralmargin = 0 
pp$ideogramheight = 0
pp$data1inmargin = -8
pp$leftmargin <- 0.1
pp$topmargin <- 14
plot.new()
kp <- plotKaryotype("hg38", plot.type=4, ideogram.plotter = NULL, labels.plotter = NULL,
                    chromosomes = chr_plot, main = "Height (covar: sex + 10PCs)", cex=1.5, plot.params = pp)
# bottom plot
kpPlotManhattan(kp, data = saige_gr, points.col="2blues", r0=autotrack(1,2, margin = 0.4), genomewideline = -log10(5e-8),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue",
                genomewide.col = "red", ymax = 8)
                #suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=5, genomewide.col = "white")
kpAddLabels(kp, labels = "Joint", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(1,2 , margin = 0.4))

# middle plot
kpPlotManhattan(kp, data = sex_gr, points.col="2blues", r0=autotrack(2,2, margin = 0.4),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=, genomewide.col = "red")
kpAddLabels(kp, labels = "Sex-Meta", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(2,2,  margin = 0.4))
kpAddChromosomeNames(kp, srt=45, cex = 1)

#invisible( dev.off() )
```

t2d / t2d age
```{r}
### joint
output_height = read.table("./saige/saige_output_t2d_t2d_age.txt", header = TRUE)
GR_input = output_height %>% mutate(chr = paste0("chr", CHR), start = POS, end = POS) %>% 
  select(chr, start, end, A1 = Allele1, A2 = Allele2, P = p.value)
saige_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)
output_height %>% arrange(p.value)

### meta-analysis
df = read.table("/hpc/group/ochoalab/tt207/meta_analysis_aim/METAL/samafs/output_metal_sex_t2d_t2d_age1.txt", header = TRUE) 
df_clean = merge(output_height %>% select(CHR, POS, MarkerName = MarkerID), df, by = "MarkerName")
GR_input = df_clean %>% 
  mutate(chr = paste0("chr", CHR), start = as.numeric(POS), end = as.numeric(POS)) %>% 
  select(-POS) %>% select(chr, start, end, A1 = Allele1, A2 = Allele2, P = P.value)
sex_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)
df_clean %>% arrange(P.value)

```

```{r  fig.width=18, fig.height=7}

#png( '/hpc/group/ochoalab/tt207/meta_analysis_aim/samafs/plots/height_man.png', width=18.5, height=13, res=1000, units = 'in')
#par(mfrow=c(5,3), mar=c(2,2,2,2))
chr_plot = c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", 
             "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22")

pp <- getDefaultPlotParams(plot.type=4)
pp$ideogramlateralmargin = 0 
pp$ideogramheight = 0
pp$data1inmargin = -8
pp$leftmargin <- 0.1
pp$topmargin <- 14
plot.new()
kp <- plotKaryotype("hg38", plot.type=4, ideogram.plotter = NULL, labels.plotter = NULL,
                    chromosomes = chr_plot, main = "T2D (covar: t2d age + sex + 10PCs)", cex=1.5, plot.params = pp)
# bottom plot
kpPlotManhattan(kp, data = saige_gr, points.col="2blues", r0=autotrack(1,2, margin = 0.4), genomewideline = -log10(5e-8),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue",
                genomewide.col = "red", ymax = 8)
                #suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=5, genomewide.col = "white")
kpAddLabels(kp, labels = "Joint", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(1,2 , margin = 0.4))

# middle plot
kpPlotManhattan(kp, data = sex_gr, points.col="2blues", r0=autotrack(2,2, margin = 0.4),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=8, genomewide.col = "red")
kpAddLabels(kp, labels = "Sex-Meta", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(2,2,  margin = 0.4))
kpAddChromosomeNames(kp, srt=45, cex = 1)
```

fast_glu / age
```{r}
### joint
output_height = read.table("./saige/saige_output_fast_glu_age.txt", header = TRUE)
GR_input = output_height %>% mutate(chr = paste0("chr", CHR), start = POS, end = POS) %>% 
  select(chr, start, end, A1 = Allele1, A2 = Allele2, P = p.value)
saige_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)
output_height %>% arrange(p.value)

### meta-analysis
df = read.table("/hpc/group/ochoalab/tt207/meta_analysis_aim/METAL/samafs/output_metal_sex_fast_glu_age1.txt", header = TRUE) 
df_clean = merge(output_height %>% select(CHR, POS, MarkerName = MarkerID), df, by = "MarkerName")
GR_input = df_clean %>% 
  mutate(chr = paste0("chr", CHR), start = as.numeric(POS), end = as.numeric(POS)) %>% 
  select(-POS) %>% select(chr, start, end, A1 = Allele1, A2 = Allele2, P = P.value)
sex_gr = makeGRangesFromDataFrame(data.frame(GR_input), keep.extra.columns = TRUE)
df_clean %>% arrange(P.value)

```

```{r  fig.width=18, fig.height=7}

#png( '/hpc/group/ochoalab/tt207/meta_analysis_aim/samafs/plots/height_man.png', width=18.5, height=13, res=1000, units = 'in')
#par(mfrow=c(5,3), mar=c(2,2,2,2))
chr_plot = c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", 
             "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22")

pp <- getDefaultPlotParams(plot.type=4)
pp$ideogramlateralmargin = 0 
pp$ideogramheight = 0
pp$data1inmargin = -8
pp$leftmargin <- 0.1
pp$topmargin <- 14
plot.new()
kp <- plotKaryotype("hg38", plot.type=4, ideogram.plotter = NULL, labels.plotter = NULL,
                    chromosomes = chr_plot, main = "Fast_glu (covar: age + sex + 10PCs)", cex=1.5, plot.params = pp)
# bottom plot
kpPlotManhattan(kp, data = saige_gr, points.col="2blues", r0=autotrack(1,2, margin = 0.4), genomewideline = -log10(5e-8),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue",
                genomewide.col = "red", ymax = 8)
                #suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=5, genomewide.col = "white")
kpAddLabels(kp, labels = "Joint", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(1,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(1,2 , margin = 0.4))

# middle plot
kpPlotManhattan(kp, data = sex_gr, points.col="2blues", r0=autotrack(2,2, margin = 0.4),
                suggestiveline = -log10(1e-5), , suggestive.col = "blue", ymax=8, genomewide.col = "red")
kpAddLabels(kp, labels = "Sex-Meta", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=2, label.margin = 0.055)
kpAddLabels(kp, labels = "-log10( p-value )", srt=90, pos=3, r0=autotrack(2,2, margin = 0.4), cex=1.5, label.margin = 0.039)
kpAxis(kp, ymin=0, ymax=8, r0=autotrack(2,2,  margin = 0.4))
kpAddChromosomeNames(kp, srt=45, cex = 1)
```